DuckDuckGo/AddressDisplayHelper.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  AddressDisplayHelper.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+extension OmniBar {
+
+    struct AddressDisplayHelper {
+
+        static func addressForDisplay(url: URL, isEditing: Bool) -> NSAttributedString {
+
+            if !isEditing, let shortAddress = shortURLString(url) {
+                return NSAttributedString(
+                    string: shortAddress,
+                    attributes: [.foregroundColor: ThemeManager.shared.currentTheme.searchBarTextColor])
+            } else {
+                return deemphasisePath(forUrl: url)
+            }
+        }
+
+        static func deemphasisePath(forUrl url: URL) -> NSAttributedString {
```

Comment:
I'm not sure we need this now do we?  The point of this was to make everything after the first slash slightly lighter but we're not showing that any more.  When the user is editing it should all be the same colour anyway imo (I don't think anyone expects the text to be demphasised while editing).

Reply:
I had mixed feelings about this part too, but ended up leaving it as it is since the text appearance was not updated while editing - that made me believe it’s intentional behavior.